presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -270,6 +284,22 @@ private Expression coerceIfNecessary(Expression original, Expression rewritten)
                 }
                 return rewritten;
             }
+
+            private Expression rewriteEnumLiteral(DereferenceExpression original, EnumType enumType)
+            {
+                Object enumValue = resolveEnumLiteral(original, enumType);
+                if (enumType instanceof LongEnumType) {
```

Comment:
Why don't we just return a `GenericLiteral` here, or if necessary, create a `Literal` that suits enum. It's a literal, why are we rewriting it to cast?